paasta_tools/mesos/master.py

```diff
@@ -69,15 +69,21 @@ def host(self):
         return "{}://{}".format(self.config["scheme"], self.resolve(self.config["master"]))
 
     @log.duration
-    def fetch(self, url, **kwargs):
+    def request(self, url, method=requests.get, **kwargs):
```

Comment:
I don't really like that you're forcing an implementation detail on the user (the implementation makes an assumption that this is a function from requests, but you can't really enforce this at the parameter level). Instead, why not make this arg a string ``http_method`` (so you'd use it like ``request(url="foo", method="POST") and do something like what's mentioned in the docs for a [prepared request](http://docs.python-requests.org/en/master/user/advanced/#prepared-requests). It keeps the interface cleaner.